tee-worker/ts-tests/common/utils.ts

```diff
@@ -447,3 +460,282 @@ export async function checkIDGraph(
+            type === 'multiple' ? context.web3Signers[index].ethereumWallet : ethereumWalletsForSign![index];
+        const data = events[index];
+        const msg = generateVerificationMessage(
+            context,
+            hexToU8a(data.challengeCode),
+            substrateSigner.addressRaw,
+            identities[index]
+        );
+        if (network === 'ethereum') {
+            const ethereumValidationData: LitentryValidationData = {
+                Web3Validation: {
+                    Evm: {
+                        message: '' as HexString,
+                        signature: {
+                            Ethereum: '' as HexString,
+                        },
+                    },
+                },
+            };
+            console.log('post verification msg to ethereum: ', msg);
+            ethereumValidationData!.Web3Validation!.Evm!.message = msg;
+            const msgHash = ethers.utils.arrayify(msg);
+            signature_ethereum = (await ethereumSigner.signMessage(msgHash)) as HexString;
+            ethereumValidationData!.Web3Validation!.Evm!.signature!.Ethereum = signature_ethereum;
+            assert.isNotEmpty(data.challengeCode, 'challengeCode empty');
+            console.log('ethereumValidationData', ethereumValidationData);
+
+            verifyDatas.push(ethereumValidationData);
+        } else if (network === 'substrate') {
+            const substrateValidationData: LitentryValidationData = {
+                Web3Validation: {
+                    Substrate: {
+                        message: '' as HexString,
+                        signature: {
+                            Sr25519: '' as HexString,
+                        },
+                    },
+                },
+            };
+            console.log('post verification msg to substrate: ', msg);
+            substrateValidationData!.Web3Validation!.Substrate!.message = msg;
+            signature_substrate = context.substrateWallet.alice.sign(msg) as Uint8Array;
+            substrateValidationData!.Web3Validation!.Substrate!.signature!.Sr25519 = u8aToHex(signature_substrate);
+            assert.isNotEmpty(data.challengeCode, 'challengeCode empty');
+        } else if (network === 'twitter') {
+            // do nothing
+            console.log('post verification msg to twitter');
+        }
+    }
+    return verifyDatas;
+}
+
+async function buildIdentityHelper(
+    address: HexString | string,
+    network: string,
+    type: string
+): Promise<LitentryIdentity> {
+    const identity: LitentryIdentity = {
+        [type]: {
+            address,
+            network,
+        },
+    };
+    return identity;
+}
+
+export async function buildIdentities(
+    network: string,
+    type: string,
+    context?: IntegrationTestContext,
+    buildTimes?: number,
+    ethereumWalletsForSign?: ethers.Wallet[]
+): Promise<LitentryIdentity[]> {
+    let identities: LitentryIdentity[] = [];
+    if (type === 'batch') {
```

Comment:
I think the types `batch` and `utility` are very confusing, it sounds like some random names and I can't figure out the intention only by reading the code.